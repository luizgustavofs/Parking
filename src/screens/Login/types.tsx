export interface LonginProps {
  setIsLogged?: true;
}
