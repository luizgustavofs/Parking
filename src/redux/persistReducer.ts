import { Reducer } from 'redux';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

const persistReducers = (reducers: Reducer<any>) => {
  const persistedReducer = persistReducer(
    {
      key: 'Parking',
      storage,
      whitelist: ['auth', 'session'],
    },
    reducers,
  );

  return persistedReducer;
};

export { persistReducers };
