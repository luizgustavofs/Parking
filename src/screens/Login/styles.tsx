import styled from 'styled-components';

export const Container = styled.div`
  display: flex;

  width: 100vw;
  height: 100vh;

  background: ${({ theme }) => theme.colors.primary};
  padding: 2rem;

  > div {
    flex: 1;
    display: flex;
    align-items: center;
    flex-direction: column;
    justify-content: center;
  }

  img {
    width: 150px;
    height: 150px;
    margin-bottom: 60px;
  }

  input {
    width: 100%;
    margin-bottom: 10px;
    border-radius: 4px;
    padding: 5px;
    text-align: center;
    border: solid 1px ${({ theme }) => theme.colors.neutral._80};

    ::placeholder {
      color: ${({ theme }) => theme.colors.neutral._70};
      font-family: 'Open Sans' sans-serif;
    }
  }

  @media (min-width: 1024px) {
    justify-content: center;

    > div {
      max-width: 300px;
      align-self: center;
    }
  }
`;
