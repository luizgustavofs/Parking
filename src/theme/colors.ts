export const colors = {
  default: '#000000',
  primary: '#4DD0E1',
  secondary: '#F2F2F2',
  tertiary: '#28DD91',
  quaternary: '#A769B2',
  error: '#FF0068',
  baseColor: '#FFFBE6',
  neutral: {
    _00: '#ffffff',
    _70: '#DADADA',
    _80: '#9B9B9B',
  },
};
