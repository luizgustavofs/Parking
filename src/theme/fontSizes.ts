import { REM_BASE } from './rem';

export const fontSizes = {
  h1: 38 / REM_BASE,
  h2: 24 / REM_BASE,
  h3: 16 / REM_BASE,
  h4: 15 / REM_BASE,
  h5: 12 / REM_BASE,
};
