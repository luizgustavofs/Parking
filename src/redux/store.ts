import { configureStore } from '@reduxjs/toolkit';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import { combineReducers } from 'redux';
import { persistStore } from 'redux-persist';

import { persistReducers } from './persistReducer';
import authReducer from './reducers/auth';
import preferencesReducer from './reducers/preferences';
import sessionReducer from './reducers/session';

const combinedReducers = combineReducers({
  auth: authReducer,
  preferences: preferencesReducer,
  session: sessionReducer,
});

export type RootReducer = ReturnType<typeof combinedReducers>;

export const store = configureStore({
  reducer: persistReducers(combinedReducers),
});

const persist = persistStore(store);

type RootState = ReturnType<typeof store.getState>;
export const useAppDispatch = () => useDispatch<typeof store.dispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
export { persist };
