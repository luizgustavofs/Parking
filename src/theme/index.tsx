import { ReactNode } from 'react';

import { ThemeProvider } from 'styled-components';

import { colors } from './colors';
import { fontSizes } from './fontSizes';
import { spaces } from './spaces';

export const theme = {
  colors,
  spaces,
  fontSizes,
};

const ProjectThemeProvider: React.FC<{ children: ReactNode }> = ({
  children,
  ...props
}) => {
  return (
    <ThemeProvider theme={theme} {...props}>
      {children}
    </ThemeProvider>
  );
};

export { ProjectThemeProvider };
