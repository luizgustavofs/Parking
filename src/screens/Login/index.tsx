/* eslint-disable @next/next/no-img-element */
/* eslint-disable jsx-a11y/alt-text */
import React, { useEffect, useState } from 'react';

import { NextPage } from 'next';
import { useTranslation } from 'react-i18next';
import { toast } from 'react-toastify';
import { useTheme } from 'styled-components';

import Button from '../../components/Button';
import { setUser } from '../../redux/reducers/auth';
import { useAppDispatch, useAppSelector } from '../../redux/store';
import { Container } from './styles';

const Login: NextPage = () => {
  const { colors } = useTheme();
  const { t } = useTranslation();
  const [username, setUsername] = useState('');
  const dispatch = useAppDispatch();
  const { user } = useAppSelector(state => state.auth);

  const handleLogin = () => {
    if (username !== '') {
      dispatch(setUser({ name: username }));
      window.location.replace('/register');
    } else {
      toast.error(t('LOGIN.ERROR'));
    }
  };

  const handleIsLogged = () => {
    if (user) {
      window.location.replace('/register');
    }
  };

  useEffect(() => handleIsLogged, []);

  return (
    <Container>
      <div>
        <img src="logo_parking.svg" />
        <input
          data-testid="username-input"
          placeholder={t('LOGIN.INPUT.USERNAME')}
          value={username}
          onChange={e => setUsername(e.target.value)}
        />
        <Button
          testID="login-test-button"
          primary
          color={colors.tertiary}
          onClick={handleLogin}>
          {t('LOGIN.BUTTON')}
        </Button>
      </div>
    </Container>
  );
};

export default Login;
