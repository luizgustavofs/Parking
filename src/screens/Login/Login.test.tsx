import { fireEvent } from '@testing-library/dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import Login from '../../pages/index';
import { store } from '../../redux/store';
import { replaceMock } from '../../tests/setupTests';
import { theme } from '../../theme';

describe('HistoricScreen', () => {
  it('should render correctly', () => {
    const { toJSON } = create(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Login />
        </ThemeProvider>
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should change screen if user add username', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Login />
        </ThemeProvider>
      </Provider>,
    );

    const input = getByTestId('username-input');
    fireEvent.change(input, { target: { value: 'gustavo' } });

    const button = getByTestId('login-test-button');
    fireEvent.click(button);

    expect(replaceMock).toHaveBeenCalledWith('/register');
  });

  it('should useEffect handle correctly', () => {
    const {} = render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Login />
        </ThemeProvider>
      </Provider>,
    );

    useEffect: jest.fn(), expect(replaceMock).not.toHaveBeenCalled();
  });
});
