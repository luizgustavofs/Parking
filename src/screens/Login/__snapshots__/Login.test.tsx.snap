// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoricScreen should render correctly 1`] = `
<div
  className="sc-gsnTZi blRkZb"
>
  <div>
    <img
      src="logo_parking.svg"
    />
    <input
      data-testid="username-input"
      onChange={[Function]}
      placeholder="LOGIN.INPUT.USERNAME"
      value=""
    />
    <button
      className="sc-bczRLJ bIDAMy"
      color="#28DD91"
      data-testid="login-test-button"
      onClick={[Function]}
    >
      <h4>
        LOGIN.BUTTON
      </h4>
    </button>
  </div>
</div>
`;
